docs/http-cloud-provider.md

```diff
@@ -0,0 +1,643 @@
+|exists|Whether the TCP Load Balancer exists or not.|true|
+Example
+
+```json
+{
+    "ingress": [
+        {
+            "ip":"127.0.0.1",
+            "hostname":"my-cloud-host1"
+        }
+    ],
+    "exists":true
+}
+```
+
+#### POST /v1/tcpLoadBalancer
+
+Creates a new tcp load balancer and return the status of the balancer.
+
+##### Sender body:
+
+|Name|Description|Example Value|
+|----|-----------|-------------|
+|loadBalancerName|The name of the TCP load balancer.|"my-tcp-balancer"|
+|region|The location of the load balancer.|"USA"|
+|externalIP|The external IP of the load balancer.|"127.0.0.1"|
+|ports|List of all the service port objects.||
+|ports:name|Name of the port. If only one port is present then this field is **optional**.|"SMTP"|
+|ports:protocol|The protocol the port works on. The value can only be either "UDP" or "TCP".|"TCP"|
+|ports:port|The port that will be exposed on the service.|1234|
+|ports:targetPort|The target port on pods selected by this service. This field is optional. It can be **string** or **int** type. If not provided, will use **ports:port** value.|1234|
+|ports:nodePort|The port on each node on which this service is exposed.|1234|
+|hosts|A list of all host names.||
+|hosts:hostname|A host name.|"my-cloud-host1"|
+|sessionAffinity|The session affinity. The value can only be either "None" or "ClientIP".|"None"|
+Example
+
+```json
+{
+    "loadBalancerName":"my-tcp-balancer",
+    "region":"USA",
+    "externalIP":"127.0.0.1",
+    "ports":[
+        {
+            "name":"SMTP",
+            "protocol":"TCP",
+            "port":1234,
+            "targetPort":1234,
+            "nodePort":1234
+        }
+    ],
+    "hosts":[
+        {
+            "hostname":"my-cloud-host1"
+        }
+    ],
+    "sessionAffinity":"None"
+}
+```
+
+##### Expected return value:
+
+|Name|Description|Example Value|
+|----|-----------|-------------|
+|ingress|List of ingress points for the load balancer.||
+|ingress:ip|The IP address of the ingress point. This is a optional field as only one of IP or hostname is required.|"127.0.0.1"|
+|ingress:hostname|The hostname of the ingress point. This is a optional field as only one of IP or hostname is required.|"my-cloud-host1"|
+|exists|Whether the TCP Load Balancer exists or not.|true|
+Example
+
+```json
+{
+    "ingress": [
+        {
+            "ip":"127.0.0.1",
+            "hostname":"my-cloud-host1"
+        }
+    ]
+}
+```
+
+#### PUT /v1/tcpLoadBalancer/{region}/{name}
+
+Updates the hosts given in the Load Balancer specified.
+
+##### Sender body:
+
+|Name|Description|Example Value|
+|----|-----------|-------------|
+|hosts|A list of all host names.||
+|hosts:hostname|A host name.|"my-cloud-host1"|
+Example
+
+```json
+{
+    "hosts":[
+        {
+            "hostname":"my-cloud-host1"
+        }
+    ]
+}
+```
+
+##### Expected return body:
+
+Send back an empty 204 response if the object was updated successfully.
```

Comment:
Just a consideration: in GCE at least, many operations are async and can take some time to complete (or fail).